openedx/core/djangoapps/content/search/documents.py

```diff
@@ -398,6 +401,8 @@ def searchable_doc_for_library_block(xblock_metadata: lib_api.LibraryXBlockMetad
 
     # Add the breadcrumbs. In v2 libraries, the library itself is not a "parent" of the XBlocks so we add it here:
     doc[Fields.breadcrumbs] = [{"display_name": library_name}]
+    # Add collections data to index if this block is part of any collection
+    doc.update(_collections_for_content_object(xblock_metadata.usage_key))
```

Comment:
@navinkarkera I think you need to call the `CONTENT_OBJECT_ASSOCIATIONS_CHANGED` event to index the [block's collections](https://github.com/openedx/edx-platform/blob/master/openedx/core/djangoapps/content/search/handlers.py#L199) in `restore_library_block` instead of indexing collections every time the library is indexed.

Reply:
@ChrisChV That is indeed better, and this led to discovery of another bug, the tags are deleted even from the database on soft deleting a component. I am currently investigating the reason and possible fix.

Reply:
> Let me know if you can think of any issue if don't delete tags from database.

I think it's okay to not delete the tags for now :+1: